buildx.go

```diff
@@ -44,6 +46,34 @@ func cmdSetupBuildx(builder Builder, driverOpts []string) *exec.Cmd {
 	if builder.BuildkitResponseHeaderTimeout != "" {
 		buildkitdFlags = append(buildkitdFlags, fmt.Sprintf("--response-header-timeout=%s", builder.BuildkitResponseHeaderTimeout))
 	}
+	if inheritAuth {
+		fmt.Println("Inheriting auth")
+		for _, env := range os.Environ() {
+			if strings.HasPrefix(env, "AWS_") {
+				parts := strings.SplitN(env, "=", 2)
+				if len(parts) == 2 {
+					envName := parts[0]
+					envValue := parts[1]
+					args = append(args, "--driver-opt", fmt.Sprintf("env.%s=%s", envName, envValue))
+				}
+			}
+		}
+
+		if tokenFilePath := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE"); tokenFilePath != "" {
+			if _, err := os.Stat(tokenFilePath); err == nil {
+				if content, err := ioutil.ReadFile(tokenFilePath); err == nil {
+					const maxSafeSize = 4 * 1024
```

Comment:
The actual buildx command string is longer, so should we keep a lower limit for the token content?